packages/grafana-runtime/src/components/DataSourcePicker.tsx

```diff
@@ -184,7 +186,7 @@ export class DataSourcePicker extends PureComponent<DataSourcePickerProps, DataS
       <div aria-label={selectors.components.DataSourcePicker.container}>
         <Select
           isLoading={isLoading}
-          disabled={disabled}
+          disabled={disabled || readOnly}
```

Comment:
I think this will make the component API confusing. Rather than introduce another prop that disables the `Select` I would expect the call site to do `<DataSourcePicker disabled={readOnly} />` or similar.

Reply:
I thought maybe the `disabled` props is currently in use, but on some page it doesn't work

Reply:
fixed